packages/material-ui/src/styles/useThemeProps.d.ts

```diff
@@ -16,7 +12,7 @@ export interface AdditionalThemeProps<Theme> {
 }
 
 export default function useThemeProps<
-  Theme extends ThemeWithProps,
```

Comment:
Why did you change this? You can use the generic where necessary.

Reply:
passing as generic seems too much work to me. the function `useThemeProps` should return the whole `theme` not just `ThemeWithProps` (in terms of type)

Reply:
I updated the code to use generic, can reconsider my opinion later when there are more similar cases.